### PR TITLE
Send headers to a page with hash segments

### PR DESCRIPTION
**Pull request by jkrems**
All the combinations of features.

### CODE REVIEW

File: lib/processes/proxy/proxy.js [hunk 1]
```diff
@@ -107,11 +111,10 @@ function proxyRequest(request, response, toPort) {
   console.log('--> %s %s', request.method, request.url);
 
   var remoteRequestOptions = buildRemoteRequestOptions(request, toPort);
-  console.log('    %j', remoteRequestOptions);
-
   if (isNewPage(request.url)) {
     modifyRequest(remoteRequestOptions, newPageOptions);
   }
+  console.log('    %j', remoteRequestOptions);
```
Comment (jkrems) [inline lib/processes/proxy/proxy.js]: Moved so the additional options are correctly logged. Makes finding issues like this much easier.

**Comment by jkrems** [comment]
Ran a decently sized test suite that uses hash urls & headers. It passes against `testium@3.3.0` with this change (didn't before):

```tap
# tests 277
# pass 277
# fail 0
```

**Comment by ageitgey** [comment]
Looks good. I guess we were just a little too hasty in the last PR.

**Comment by jkrems** [comment]
v1.1.4

**Pull request merged**